src/Models/AuditLog.cs

```diff
@@ -0,0 +1,10 @@
+namespace IdentityServerConfig.Models;
+
+public class AuditLog
+{
+    public int ID { get; set; }
+    public string UserId { get; set; }
+    public string UserName { get; set; }
+    public DateTime Timestamp { get; set; }
+    public string Action { get; set; }
```

Comment:
I like the overall design of this feature so kudos for that.

I'm thinking only having an action like "ReferenceTokenChecked" might not be enough though to understand what has happened. Especially for more complicated actions like revoking a reference token might require additional context, such as the client for which it was revoked. Perhaps this could be generalised a bit by allowing the audit log to store arbitrary events, possibly represented as an `IAuditableEvent` interface with at least a `string Action` (or maybe `Type` would be more appropriate then) with potentially additional properties like ClientId. These could then be stored as JSON in the database.

Reply:
added in 9296551